addons/Bonanza/Bonanza.lua

```diff
@@ -0,0 +1,195 @@
+--[[
+Copyright © 2018, from20020516
+All rights reserved.
+
+Redistribution and use in source and binary forms, with or without
+modification, are permitted provided that the following conditions are met:
+
+    * Redistributions of source code must retain the above copyright
+      notice, this list of conditions and the following disclaimer.
+    * Redistributions in binary form must reproduce the above copyright
+      notice, this list of conditions and the following disclaimer in the
+      documentation and/or other materials provided with the distribution.
+    * Neither the name of checkparam nor the
+      names of its contributors may be used to endorse or promote products
+      derived from this software without specific prior written permission.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+DISCLAIMED. IN NO EVENT SHALL from20020516 BE LIABLE FOR ANY
+DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+(INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+(INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.]]
+
+_addon.name = 'Bonanza'
+_addon.author = 'from20020516'
+_addon.version = '1.1'
+_addon.command = 'bonanza'
+
+extdata = require('extdata')
+packets = require('packets')
+require('logger')
+
+marble = {id=2559,price=2000,limit=10} --general settings
+
+win_numbers = {}
+math.randomseed(os.time())
+
+--search your bags for bonanza marble and get number then add prefix 0
+function search_marbles()
+  local marbles = {}
+  local bags = {0,1,2,4,5,6,7,9}
+  for bag_ind=1,#bags do
+    for index=1,windower.ffxi.get_bag_info(bags[bag_ind]).max do
+      local item = windower.ffxi.get_items(bags[bag_ind],index)
+      if item.id == marble.id then --bonanza marble
+        local five_digits = string.format("%05s",extdata.decode(item).number)
+        table.insert(marbles,{bags[bag_ind],index,five_digits})
+      end
+    end
+  end
+  return marbles;
+end
+
+windower.register_event('addon command',function(...)
+  moogle = windower.ffxi.get_mob_by_name('Bonanza Moogle')
+  decides = {}
+  local cmd={...}
+  local count = marble.limit-#search_marbles()
+
+  for i=1,#cmd do
+    math.random() --shuffle rand
+    cmd[i] = tonumber(cmd[i]) and cmd[i]*1 or cmd[i]
+  end
+
+  if cmd[1] == 'judge' then
+    windower.send_command('input /smes')
+  elseif count == 0 then
+    error('you have already '..marble.limit..' marbles.')
+  elseif cmd[1] == 'random' then
+    for i=1,count do
+      table.insert(decides,math.random(0,99999))
```

Comment:
You should be aware that `math.random` relies on the C `rand` function which can only generate integers between 0 and ‭32767‬ (inclusive). Lua effectively does `rand() / 32767.0 * (upper - lower) + lower`. What this means is that some values, e.g. 48831, will never be possible.

I suggest instead calling `math.random` multiple times and combining the results.
```lua
table.insert(decides,math.random(0,99) + math.random(0,99)*100 + math.random(0,9)*10000)
```